source/common/upstream/load_balancer_impl.cc

```diff
@@ -775,6 +845,24 @@ void EdfLoadBalancerBase::refresh(uint32_t priority) {
   }
 }
 
+bool EdfLoadBalancerBase::noHostsAreInSlowStart() {
+  auto current_time = std::chrono::duration_cast<std::chrono::milliseconds>(
+      time_source_.systemTime().time_since_epoch());
+  // todo(nezdolik) extract this as a field
+  auto slow_start_window_ms = std::chrono::milliseconds(slow_start_window * 1000);
+  // First check if any host is within slow start window. This condition holds if at least latest
+  // added host is within slow start window.
+  // If all hosts are out of the window, we no longer need to track them and therefore we erase
+  // tracked hosts set.
+  if (current_time - latest_host_added_time > slow_start_window_ms) {
+    hosts_in_slow_start_.erase(hosts_in_slow_start_.begin(), hosts_in_slow_start_.end());
+    return true;
+  } else {
+    // Second, check if there is any hosts in tracked hosts set.
+    return hosts_in_slow_start_.size() == 0;
```

Comment:
Can you just see if this is empty at the beginning of the function and return true if empty?

Reply:
taken care of